homeassistant/components/sensorpush_cloud/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for the SensorPush Cloud integration."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+)
+
+from .api import SensorPushCloudApi, SensorPushCloudError
+from .const import CONF_DEVICE_IDS, DOMAIN, LOGGER
+
+
+class SensorPushCloudConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for SensorPush Cloud."""
+
+    VERSION = 1
+
+    api: SensorPushCloudApi | None
+    data: dict[str, Any]
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.data = {}
+
+    async def async_step_configure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the configuration step."""
+        if TYPE_CHECKING:
+            assert self.api is not None
+        if user_input is not None:
+            if not user_input[CONF_DEVICE_IDS]:
+                return self.async_abort(reason="no_devices_selected")
+            self.data |= user_input
+            return self.async_create_entry(title=self.data[CONF_EMAIL], data=self.data)
+
+        errors: dict[str, str] = {}
+        options: list[SelectOptionDict] = []
+        try:
+            sensors = await self.api.async_sensors()
+            options = [{"value": k, "label": v["name"]} for k, v in sensors.items()]
+        except SensorPushCloudError as e:
+            errors["base"] = str(e)
+        except Exception:  # noqa: BLE001
+            LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            if not options:
+                errors["base"] = "no_devices_found"
+
+        return self.async_show_form(
+            step_id="configure",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_DEVICE_IDS, default=[]): SelectSelector(
+                        SelectSelectorConfig(options=options, multiple=True)
+                    )
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            email, password = user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+            self.api = SensorPushCloudApi(self.hass, email, password)
+            await self.async_set_unique_id(email, raise_on_progress=False)
```

Comment:
Can you change the email?

Reply:
Nope, this is a fixed part of the SensorPush API used for login. Once you register for API access, it is fixed.

Reply:
What happens if you need to change your email address? Do you need to create a new account?

Reply:
I just double checked and it looks like you can change the email using the dashboard. In this case, could it make sense to have a follow-up PR that supports the reauth config flow and allow the user to change both email and password?